xen-bugtool

```diff
@@ -1634,8 +1636,65 @@ def removeNoError(filename):
     except OSError:
         pass
 
-class TarOutput:
+class TarSubArchive(io.BytesIO):
+    """Utility class for adding subarchives to an ZIP or tar output archive"""
+
+    def __init__(self, basepath, tar_filename):
+        """
+        Create the object, defining the base path of the files and the tar file name.
+        :param basepath: path below which all files shall be captured into the subarchive
+        :param tar_filename: Name of the tar file in the parent archive
+        """
+        self.basepath = basepath
+        self.mtime = time.time()
+        self.name = tar_filename
+        self.file = tarfile.open(fileobj=self, mode="w|", dereference=True)
+
+    def add(self, name, filename):
+        """
+        Add a file to the subarchive
+        :param name: Recorded path of the the file in the tar archive for extraction
+        :param filename: Real file name of the file to be added to the tar archive
+        """
+        self.file.addfile(self.file.gettarinfo(filename, name), open(filename, "rb"))
+
+class ArchiveWithTarSubarchives(object):
+    """Base class for TarOutput and ZipOutput with support to create sub-archives"""
+
+    def __init__(self):
+        """Initialize the defined sub-archives to be an empty list"""
+        self.subarchives = []
+
+    def declare_subarchive(self, basepath, tar_filename):
+        """
+        Declare a subarchive by defining the base path of the files and the tar file name.
+        :param basepath: path below which all files shall be captured into the subarchive
+        :param tar_filename: Name of the tar file in the parent archive
+        """
+        self.subarchives.append(TarSubArchive(basepath, tar_filename))
+
+    def addFileToSubarchive(self, name, filename):
+        """If filename belongs to a subarchive, add the file to it and return True"""
+        for subarchive in self.subarchives:
+            if filename.startswith(subarchive.basepath):
+                subarchive.add(name, filename)
+                return True
+        return False
+
+    def add_subarchives(self):
+        """Close all subarchives and add the archive files to the Tar/Zip output"""
+        for subarchive in self.subarchives:
+            if len(subarchive.file.getmembers()):
+                subarchive.file.close()
+                self.addFile(subarchive.name, subarchive)
+
+    def addFile(self, _name, _data):
```

Comment:
These methods are using different naming conventions. Lower case with underscores is preferred by PEP8.

Reply:
Done, addFile() is a method in `ZipOutput` and `TarOutput` as well, which this abstract base method of the base class has to match.

But, as the name addFile() is quite ambiguous for what it does, even wrong, because it does not add a file but data at a given path, I'm happy to rename it and all its callers, which I have done as a result. Looks much better!